executor/window.go

```diff
@@ -174,3 +175,130 @@ func (e *WindowExec) copyChk(chk *chunk.Chunk) {
 		chk.MakeRefTo(i, childResult, col.Index)
 	}
 }
+
+// windowProcessor is the interface for processing different kinds of windows.
+type windowProcessor interface {
+	// consumeGroupRows updates the result for an window function using the input rows
+	// which belong to the same partition.
+	consumeGroupRows(ctx sessionctx.Context, rows []chunk.Row) ([]chunk.Row, error)
```

Comment:
This function is unnecessary?
Argument `rows` of `appendResult2Chunk` contains all rows which belong to the same partition, so we can do `UpdatePartialResult` at there once?